File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        // result 
        int[] result = new int[nums.length];
        for(int i=0;i<nums.length;i++) {
            int value = 1;
            for(int j=0;j<nums.length;j++) {
                if(i!=j) {
                    value = value * nums[j];
                }
            }
            result[i] = value;
        }
        return result;
    }
}  

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        curr,res = [],[]
        self.helper(nums, curr, res)
        return res


    def helper(self, nums, curr, res):
        if len(curr) == len(nums):
            res.append(curr.copy())
            return 

        for i in nums:
            if i not in curr:
                curr.append(i)
                self.helper(nums, curr, res)
                curr.pop()

        